src/smartcard/util/__init__.py

```diff
@@ -210,61 +203,21 @@ def toHexString(data=[], format=0):
     '0X3B, 0X65, 0X00, 0X00, 0X9C, 0X11, 0X01, 0X01, 0X03'
     """
 
-    for byte in tuple(data):
-        pass
-
-    if type(data) is not list:
+    if not (data is None or isinstance(data, list)):
         raise TypeError('not a list of bytes')
 
-    if data is None or data == []:
+    if not data:
         return ""
-    else:
-        pformat = "%-0.2X"
-        if COMMA & format:
-            separator = ","
-        else:
-            separator = ""
-        if not PACK & format:
-            separator = separator + " "
-        if HEX & format:
-            if UPPERCASE & format:
-                pformat = "0X" + pformat
-            else:
-                pformat = "0x" + pformat
-        return (separator.join(map(lambda a: pformat % ((a + 256) % 256), data))).rstrip()
-
-
-# FIXME This appears to duplicate toASCIIString()
-def HexListToBinString(hexlist):
```

Comment:
Do NOT remove functions that are part of the documented API
Maybe we should deprecate them first.
I found https://pypi.org/project/Deprecated/ but it is not in the Python 3 standard lib.

Reply:
I've updated the PR to follow this. Python support `DeprecationWarning` for this purpose.